tests/integration/helpers/0_common_instance_config.xml

```diff
@@ -25,5 +25,6 @@
         <select_from_system_db_requires_grant>true</select_from_system_db_requires_grant>
         <select_from_information_schema_requires_grant>true</select_from_information_schema_requires_grant>
         <settings_constraints_replace_previous>true</settings_constraints_replace_previous>
+        <role_cache_expiration_time>6</role_cache_expiration_time>
```

Comment:
Why 6?

Reply:
The number must be neither too large (otherwise test would be slow) nor too small (to avoid race alike conditions, where it is not clear what we are actually testing).
I've reduced values, now expiration time in test environment is 2 seconds (not 6) and delay in test is 5 seconds (not 10).